pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/notPepperKit/NotPepperKitCreationService.java

```diff
@@ -0,0 +1,238 @@
+        DDPInstanceDto ddpInstanceDto = maybeDdpInstanceDto.get();
+        //getting the instance with isHasRole being set to true if the instance has role juniper_study
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole(ddpInstanceDto.getInstanceName(), "juniper_study");
+
+        if (!ddpInstance.isHasRole()) {
+            throw new RuntimeException("This is not a Juniper study!");
+        }
+        HashMap<String, KitType> kitTypes = KitType.getKitLookup();
+        String key = kitTypeName + "_" + ddpInstanceDto.getDdpInstanceId();
+        KitType kitType = kitTypes.get(key);
+        if (kitType == null) {
+            return new KitResponse(UNKNOWN_KIT_TYPE, juniperKitRequest.getJuniperKitId());
+        }
+
+        Map<Integer, KitRequestSettings> kitRequestSettingsMap =
+                KitRequestSettings.getKitRequestSettings(String.valueOf(ddpInstanceDto.getDdpInstanceId()));
+        KitRequestSettings kitRequestSettings = kitRequestSettingsMap.get(kitType.getKitTypeId());
+
+        // if the kit type has sub kits > like for testBoston
+//        boolean kitHasSubKits = kitRequestSettings.getHasSubKits() != 0;
+
+        log.info("Setup EasyPost...");
+        EasyPostUtil easyPostUtil = new EasyPostUtil(ddpInstanceDto.getInstanceName());
+
+        if (!checkAddress(juniperKitRequest, kitRequestSettings.getPhone(), easyPostUtil)) {
+            return new KitResponse(ADDRESS_VALIDATION_ERROR, juniperKitRequest.getJuniperKitId());
+        }
+
+        ArrayList<KitRequest> orderKits = new ArrayList<>();
+
+
+        TransactionWrapper.inTransaction(conn -> {
+            createKit(ddpInstance, kitType, juniperKitRequest, kitRequestSettings, easyPostUtil, kitTypeName,  orderKits, conn);
+
+//            only order if external shipper name is set for that kit request, not needed for now
+//            if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+//                return orderExternalKits(kitRequestSettings, orderKits, easyPostUtil, shippingCarrier, conn);
+//            }
+            return null;
+        });
+
+        //send not valid address back to client
+        log.info(juniperKitRequest.getJuniperKitId() + " " + ddpInstance.getName() + " " + kitTypeName + " kit created");
+        return new KitStatus(null, null);
+    }
+
+    /**
+     * checkAddress tries creating an address in EasyPost. If it is successful,
+     * sets the kit's easypostAddressId and returns true, if not returns false
+     * An address is valid only if participant has shortId, first - and lastName, for Juniper shortId is the juniperParticipantId
+     *
+     * @param juniperKitRequest the JuniperKitRequest with address to check
+     */
+
+    public boolean checkAddress(JuniperKitRequest juniperKitRequest, String phone, EasyPostUtil easyPostUtil) {
+        if ((StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID()) && StringUtils.isBlank(juniperKitRequest.getExternalOrderNumber()))
```

Comment:
Can you comment on what this test is doing? If we have a participant ID we are ok with not having this other info?
Looks like we use (at least) last name below.

Reply:
of course, it checks if we have some sort of a kit id (or external id) at this point, and it also checks that there is a last name. It will fail if there is no last name or if there is no id.
However, I will remove the external order check because they don't use external shippers anyways and we don't want to make the mistake of passing the condition with an id that is not used